### PR TITLE
Fix toEnum for (Maybe a)

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

File: src/Data/Enum.purs [hunk 1]
```diff
@@ -196,8 +196,8 @@ instance boundedEnumMaybe :: BoundedEnum a => BoundedEnum (Maybe a) where
   toEnum = to cardinality
     where
     to :: Cardinality a -> Int -> Maybe (Maybe a)
-    to _ 0 = Nothing
-    to (Cardinality ca) n | n <= ca = Just $ toEnum (n - 1)
+    to _ 0 = Just Nothing
+    to (Cardinality ca) n | n > 0 && n <= ca = Just $ toEnum (n - 1)
```
Comment (paf31) [inline src/Data/Enum.purs]: Why are we even doing a check here? Why not let the call to `toEnum` fail?

```purescript
to 0 = pure Nothing
to n = Just <$> toEnum (n - 1)
```

**Comment by jacereda** [comment]
@paf31 much better :)

**Comment by paf31** [comment]
Thanks! @garyb, any comments on this? Looks good to me.

**Comment by garyb** [comment]
Nope, looks great!

**Pull request merged**